src/poc/miner_poc_statem.erl

```diff
@@ -114,10 +114,18 @@ init(Args) ->
         {ok, _State, #data{poc_restarts=POCRestarts}} when POCRestarts == 0 ->
             {ok, requesting, #data{base_dir=BaseDir, blockchain=Blockchain,
                                    address=Address, poc_interval=Delay, state=requesting}};
+        %% we loaded a state from the saved statem file, initialise with this if its an exported/supported state
         {ok, State, #data{poc_restarts = POCRestarts} = Data} ->
-            {ok, State, Data#data{base_dir=BaseDir, blockchain=Blockchain,
-                                  address=Address, poc_interval=Delay, state=State,
-                                  poc_restarts = POCRestarts - 1}}
+            case is_supported_state(State) of
+                true ->
+                    {ok, State, Data#data{base_dir=BaseDir, blockchain=Blockchain,
+                                          address=Address, poc_interval=Delay, state=State,
+                                          poc_restarts = POCRestarts - 1}};
+                false ->
+                    lager:debug("Loaded unsupported state ~p, ignoring and defaulting to requesting", [State]),
+                    {ok, requesting, #data{base_dir=BaseDir, blockchain=Blockchain,
```

Comment:
Do you need to reset all the variables in the #data record?

Reply:
state specific variables are set via the state enter handlers, so at this point only need to set global vars.

Reply:
Ah I see.. thought you were needlessly assigning here